zest/releaser/towncrier_hooks.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+from zest.releaser import utils
+import logging
+import os
+import pkg_resources
+import toml
+
+# towncrier might become a conditional dependency of zest.releaser,
+# for example in the recommended list.
+# If we split this into a separate package, it should be a hard dependency.
+try:
+    pkg_resources.get_distribution('towncrier')
+except pkg_resources.DistributionNotFound:
+    HAS_TOWNCRIER = False
+else:
+    HAS_TOWNCRIER = True
+
+
+logger = logging.getLogger('towncrier_hooks')
+TOWNCRIER_MARKER = '_towncrier_applicable'
+TOWNCRIER_CONFIG_FILE = 'pyproject.toml'
+
+
+def load_config():
+    fn = os.path.join(TOWNCRIER_CONFIG_FILE)
+    if not os.path.exists(fn):
+        return
+    with open(fn, 'r') as conffile:
+        config = toml.load(conffile)
+    if 'tool' not in config:
+        return
+    config = config['tool']
+    if 'towncrier' not in config:
+        return
+    config = config['towncrier']
+    # Currently we need 'package' in the config.
+    if 'package' not in config:
+        logger.error(
+            "The [tool.towncrier] section of %s "
+            "has no required 'package' key.", TOWNCRIER_CONFIG_FILE)
+        return
+    return True
+
+
+def _check_towncrier():
+    # First check if the towncrier tool is available.
+    # towncrier might be on the PATH but not importable or the other way around.
+    # So could be tricky to know for sure that it is available.
+    # Maybe call 'towncrier --help' and see if that gives an error.
+    if not HAS_TOWNCRIER:
+        return False
+    # Read pyproject.toml with the toml package.
+    if not load_config():
+        return False
+    return True
+
+
+def check_towncrier(data):
```

Comment:
The names `_check_towncrier()` and `check_towncrier()` are very similar. The underscore version should perhaps be named something like `_is_towncrier_enabled()`?